x.zig

```diff
@@ -2478,8 +2541,171 @@ pub const ConnectSetup = struct {
     pub fn getScreensPtr(self: @This(), format_list_limit: u32) [*]align(4) Screen {
         return @alignCast(@ptrCast(self.buf.ptr + format_list_limit));
     }
+    pub fn getScreens(self: @This(), allocator: std.mem.Allocator) ![]align(4) *Screen {
+        const format_list_offset = ConnectSetup.getFormatListOffset(self.fixed().vendor_len);
+        const format_list_limit = ConnectSetup.getFormatListLimit(format_list_offset, self.fixed().format_count);
+
+        const screen_count = self.fixed().root_screen_count;
+        var screens = try allocator.alloc(*Screen, screen_count);
+        var pointer_offset = format_list_limit;
+        for (0..screen_count) |screen_index| {
+            var screen: *align(4) Screen = @alignCast(@ptrCast(self.buf.ptr + pointer_offset));
+            screens[screen_index] = screen;
+            const depths = try screen.getAllowedDepths(allocator);
+            for (depths) |depth| {
+                pointer_offset += @sizeOf(ScreenDepth) + (@sizeOf(VisualType) * depth.visual_type_count);
+            }
+            pointer_offset += @sizeOf(Screen);
+        }
+
+        return screens;
+    }
 };
 
+var TEST_RECEIVED_CONNECT_SETUP_BUFFER align(4) = [_]u8{
```

Comment:
Some dummy data of 2x screens that have depths at 1, 4, 24, 32 and some visual types for each depth.

Reply:
Added a comment to the diff :fast_forward:

Reply:
Oh cool a unit test for a connection. Let's put this in a separate file though.  Maybe we could add a `test` directory and put it in there?  Or we could just add a `test.zig`?

Reply:
Moved the tests to a `tests/` directory and things seem to run as expected when using `zig build test --summary all`. Does that look about right?